### DISCUSSION TITLE
Persistence unit tests. How to wait until all events are saved to journal?

### DISCUSSION THREAD

**Discussion by AlexeyRokhin** [akkadotnet/akka.net#7094] [Q&A]
Hi.

I'm creating unit tests to test persistence and recovery. To collect saved and recovered messages I'm using a JournalInterceptor that stores the persistence events in the `List<IPersistentRepresentation> Messages` property.

So, I send a message to the actor using the following code:
```csharp
var myResponse = await myActor.Ask<MyResponse>(new MyRequest());
```
Then I try to validate the number of persistence messages and the messages themselves:
```csharp
Assert.Equal(3, this.writeInterceptor.Messages.Count);
Assert.True(this.writeInterceptor.Messages[0].Payload is ...);
...
```
And `Assert.Equal` fails because `this.writeInterceptor.Messages.Count` actually equals to `1`.
When I add `await Task.Delay(100)` before the validation, then it succeeds.

Then I stop the current actor and create a new one with the same `PersistenceId`. And I get the same issue with `recoveryInterceptor`.
Without `await Task.Delay(100)` its `Messages` collection is empty.

Is there any way to fix the issues without using `Task.Delay`?

**Comment by jaydeboer** [comment]
@AlexeyRokhin Would using `AwaitAssert` work for you?
``` csharp
AwaitAssert(() => Assert.Equal(3, this.writeInterceptor.Messages.Count));
```

**Reply by AlexeyRokhin** [reply]
@jaydeboer 
Yes, that works. Thank you!
Any other ways?
